tests/blackbox_test.go

```diff
@@ -184,28 +198,87 @@ func outer(t *testing.T, test types.Test, negativeTests bool) {
 		// Finish data setup
 		for _, part := range disk.Partitions {
 			if part.GUID == "" {
-				part.GUID = generateUUID(t, ctx)
+				part.GUID, err = generateUUID(t, ctx)
+				if err != nil {
+					return err
+				}
+			}
+			err := updateTypeGUID(t, part)
+			if err != nil {
+				return err
 			}
-			updateTypeGUID(t, part)
 		}
 
 		disk.SetOffsets()
 		for _, part := range test.Out[i].Partitions {
-			updateTypeGUID(t, part)
+			err := updateTypeGUID(t, part)
+			if err != nil {
+				return err
+			}
 		}
 		test.Out[i].SetOffsets()
 
 		// Creation
-		createVolume(t, ctx, i, disk.ImageFile, imageSize, 20, 16, 63, disk.Partitions)
-		disk.Device = setDevices(t, ctx, disk.ImageFile, disk.Partitions)
+		err = createVolume(t, ctx, i, disk.ImageFile, imageSize, 20, 16, 63, disk.Partitions)
+		if err != nil {
+			return err
+		}
+		// Move value into the local scope, because disk.ImageFile will change
+		// by the time this runs
+		imageFile := disk.ImageFile
+		defer func() {
```

Comment:
Same as earlier, we could create the file & truncate it to the full size but fail the partition table creation and then error out and leave this not cleaned up.